templates/default/deploy-scripts/src/deployer.rs

```diff
@@ -0,0 +1,459 @@
+use crate::v2::BlockIdentifier::Best;
+use concordium_rust_sdk::{
+    common::types::TransactionTime,
+    id::types::AccountAddress,
+    smart_contracts::common::{Address, ModuleReference},
+    types::{
+        hashes::{HashBytes, TransactionMarker},
+        queries::AccountNonceResponse,
+        smart_contracts::{ContractContext, InvokeContractResult, WasmModule},
+        transactions::{
+            self,
+            send::{deploy_module, init_contract, GivenEnergy},
+            InitContractPayload, UpdateContractPayload,
+        },
+        AccountTransactionEffects, BlockItemSummary, BlockItemSummaryDetails, ContractAddress,
+        Energy, RejectReason, TransactionType, WalletAccount,
+    },
+    v2,
+};
+use std::path::Path;
+
+use crate::DeployError;
+
+/// A struct representing the deployed module on the chain.
+#[derive(Clone, Debug)]
+pub struct ModuleDeployed {
+    /// Module reference on the chain.
+    pub module_ref: ModuleReference,
+}
+
+/// A struct representing a smart contract instance on the chain.
+#[derive(Clone, Debug)]
+pub struct ContractInitialized {
+    /// Smart contract address on the chain.
+    pub contract: ContractAddress,
+}
+
+/// A struct containing connection and wallet information.
+#[derive(Debug)]
+pub struct Deployer {
```

Comment:
I am not sure that this struct makes a lot of sense in the first place, maybe just provide a bunch of functions which takes the client and keys as well

Reply:
I agree that currently, this struct does not hold a lot of information but it could contain many more settings in the future when we add additional functionalities. I see its value in combining/bundling all these setup settings. E.g. possible additional settings that this struct can hold: 

- several account keys to deploy with different accounts
- different network settings 
- dry-run settings (e.g. the script first dry-runs the deployment before executing it live on mainnet)